### PR TITLE
Add support for apply on merge in atlantis

### PR DESCRIPTION
**Pull request by satyamz**
## What:
Add support for apply on merge in atlantis. As a part of this PR:

- Add apply requirement as `merged`.
- Add pull event for detecting `merged` PRs and applying the change via atlantis automatically.
- Update go generate mocks and unit tests.

## Why:

Atlantis needs `atlantis apply` comment command to trigger apply workflow. We wanted to atlantis to be able to apply the plan after PR is merged. More information: https://github.com/runatlantis/atlantis/issues/2172

### CODE REVIEW

File: server/events/vcs/azuredevops_client.go [hunk 1]
```diff
@@ -217,6 +217,11 @@ func (g *AzureDevopsClient) PullIsMergeable(repo models.Repo, pull models.PullRe
 	return true, nil
 }
 
+// PullIsMerged returns true if the merge request is merged.
+func (g *AzureDevopsClient) PullIsMerged(repo models.Repo, pull models.PullRequest, vcsstatusname string) (bool, error) {
+	return false, errors.New("feature is not supported for BitBucket cloud merge request")
```
Comment (jacekn) [inline server/events/vcs/azuredevops_client.go:222]: Is "BitBucket" a typo?

**Review by jacekn — COMMENTED** [review]
I dropped a few small comments inline

File: server/controllers/events/events_controller.go [hunk 2]
```diff
@@ -462,6 +464,25 @@ func (e *VCSEventsController) handlePullRequestEvent(logger logging.SimpleLoggin
 		return HTTPResponse{
 			body: "Pull request cleaned successfully",
 		}
+	case models.MergedPullEvent:
+		//If the pull request was merged, we check if apply-on-merge requirement was specified.
+		//TODO: Implement apply plan here.
```
[the remark this answers is not in the document]
Comment (satyamz) [reply server/controllers/events/events_controller.go:469]: No. Will remove it.

File: server/events/vcs/azuredevops_client.go (hunk 1 above)
Comment (satyamz) [reply server/events/vcs/azuredevops_client.go:222]: Yes. It should be AzureDevops

**Review by jacekn — APPROVED** [review]

**Pull request merged**